File: pythonProject/proj/Plant/Plant.py
import proj.myOrganizm
from random import *
from proj.myOrganizm import Point
from proj.myWorld import World
class Plant(proj.myOrganizm.Organizm):
    def __init__(self, world: World, point: Point, power=0, initiative=0, age=0):

        super().__init__(world, point, power, initiative, age)
        self.isAnimal = True
    def Kolizja(self, org):
        self._world.deleteOrganizm(self)


    def Akcja(self):
        a = randint(0, 100)
        if (a > 93):
            self.growth()

    def growth(self):
        a = randint(0, 3)
        newTemp = self._point
        if(a==0):
            if (self._point.getY() - 1 > 0):
                newTemp = Point(self._point.getX(), self._point.getY() - 1)
                
        elif(a==1):
            if (self._point.getX() - 1 > 0):
                newTemp = Point(self._point.getX() - 1, self._point.getY())
           

        elif(a==2):
            if (self._point.getX() + 2 < self._world.getX()):
                newTemp = Point(self._point.getX() + 1, self._point.getY())
        elif(a==3):
            if (self._point.getY() + 2 < self._world.getY()):
                newTemp = Point(self._point.getX(), self._point.getY() + 1)
        
        
        if (self._world.whoIs(newTemp, newTemp) == None):
            self._world.createOrganizm(self.getTyp(), newTemp)

File: javaProject/src/main/java/project_java/Plant.java
package project_java;

import java.awt.*;
import java.util.Random;

public abstract class Plant extends Organizm{
    public Plant(World world,Point position, int age, int power, int inicjatywa){
        super(world,position,age,power,inicjatywa);
        inicjatywa = 0;
        isAnimal = false;
    }
    public Plant(World world, Point position){
        super(world, position);
    }
    @Override
    public Color getColor() {
        return null;
    }

    @Override
    public void Akcja() {
        Random random = new Random();
        int a = random.nextInt(100);
        if(a>93){
            growth();
        }
    }

    @Override
    public void Kolizja(Organizm org) {
        world.deleteOrganizm(this);
    }

    @Override
    public String getName() {
        return null;
    }

    @Override
    public Image getImage() {
        return null;
    }

    public void growth() {
        Random random = new Random();
        int a = random.nextInt(4);
        Point newTemp = this.position;
        switch (a) {
            case 0:
                if (position.getY() - 1 > 0 )
                newTemp = new Point(position.getX(), position.getY() - 1 );
                break;
            case 1:
                if (position.getX() - 1 > 0)
                newTemp = new Point(position.getX()-1, position.getY()  );
                break;
            case 2:
                if (position.getX() + 2 < world.getX())
                    newTemp = new Point(position.getX()+1, position.getY()  );
                break;
            case 3:
                if (position.getY() + 2 < world.getY())
                newTemp = new Point(position.getX(), position.getY() +1 );
                break;
            default:
                break;
        }
        if (world.whoIs(newTemp) == null) {
            world.createOrganizm(this.getTyp(), newTemp);

        }


    }
}
